packages/cli/src/deployer/interfaces.ts

```diff
@@ -0,0 +1,91 @@
+import * as path from "path";
+import * as fs from "fs";
+import { sync as globSync } from "glob";
+import { CloudDirectorConfig, DefinedInterfaceApi, DefinedInterfaceBehaviorsApi } from '@vcd/node-client';
+import { ObjectDeployer } from './ObjectDeployer';
+
+const getIdComponent = (det: any): string => {
+    return `${det.vendor}:${det.nss}:${det.version}`
+}
+
+export class InterfacesDeployer implements ObjectDeployer {
+    intApi: DefinedInterfaceApi
+    behApi: DefinedInterfaceBehaviorsApi
+
+    constructor(private apiConfig: CloudDirectorConfig, private rootDir: string, private debug: (...args: any[]) => void) {
+        this.intApi = this.apiConfig.makeApiClient(DefinedInterfaceApi)
+        this.behApi = this.apiConfig.makeApiClient(DefinedInterfaceBehaviorsApi)
+    }
+
+    private async traverse(visitor: (definedInterface: any, existingInt?: any) => Promise<any>) {
+        const files = globSync(path.resolve(this.rootDir, 'lib', 'interfaces', "**/*.json"))
+        const serverEntities = await this.intApi.queryInterfaces(1, 100)
+        const existingInts = serverEntities.body.values
+            .reduce((prev: any, curr) => {
+                prev[getIdComponent(curr)] = curr
+                return prev
+            }, {})
+        this.debug(existingInts)
+        return Promise.all(
```

Comment:
Just a note. If you have huge list of files this could potentially throw OOM Exception.

Reply:
This is a temporary implementation, since with the package support it will be done differently. On the other hand we don't need to support huge list of file I think, or at least not at this point.